### DISCUSSION TITLE
How to redirect to a new route, only on success of a REST API Call?

### DISCUSSION THREAD

**Discussion by RajaTheKing826** [remix-run/remix#3466] [Q&A]
I want to redirect to another route only after successful completion of the REST Api that i have written in action. In the docs i have seen that they are simply redirecting irrespective of the API call success, but in my case i want redirection only on success of API that i have written

**Comment by sergiodxa** [comment]
You need to await your REST API call and return a redirect after that
```ts
export let action: ActionFunction = async () => {
  try {
    await doYourApiRequest()
    return redirect("/somewhere/else") // this will happen after doYourApiRequest
  } catch (error) {
    // this will happen if doYourApiRequest failed
    throw json({ message: error.message }, 400)
  }
}
```
